src/django/api/serializers.py

```diff
@@ -632,6 +640,68 @@ def get_affiliation_choices(self, claim):
     def get_certification_choices(self, claim):
         return FacilityClaim.CERTIFICATION_CHOICES
 
+    def get_product_type_choices(self, claim):
+        seeds = [
+            seed
+            for seed
+            in ProductType.objects.all().values_list('value', flat=True)
+            or []
+        ]
+
+        new_values = FacilityClaim \
+            .objects \
+            .all() \
+            .values_list('facility_product_types', flat=True)
+
+        if new_values.exists():
+            new_values = new_values[0]
+
+        values = [
+            new_value
+            for new_value
+            in new_values or []
+        ]
```

Comment:
I'd like to check my reasoning here a little bit:

- the `values_list` call for `seeds` will always return a list because of our data migration
- the `values_list` call for `facility_product_types` will return `None` if no custom types have been entered
- the `values_list` call for `facility_product_types returns a list of lists when it does return values

So we check whether `new_values.exists()` here to see whether it is not `None` and if it is not, we remove one level of nesting by getting the 0th index. If it is `None`, that's okay because the `or []` thing in the list comprehension above will just make `values` into an empty list.

Subsequently we union the seeds and custom values together before making the tuples to send back as choices.

(Same logic for production types, btw). Does this seem reasonable? I tested it out and it worked both when there were no existing `facility_product_type` values and when there were some existing values.

Reply:
I tested this reasoning by adding products to 2 different profiles. It does not look like we want `new_values[0]`

```
>>> new_values = FacilityClaim.objects.all().values_list('facility_product_types', flat=True)
>>> new_values
<QuerySet [['Bedskirts', 'Foo'], ['Chaps', 'Foo', 'Spats']]>
>>> new_values[0]
['Bedskirts', 'Foo']
```

We can combine set and itertools.chain to get a set that we can union with our seeds.
```
>>> from itertools import chain
>>> set(chain.from_iterable(new_values))
{'Bedskirts', 'Foo', 'Spats', 'Chaps'}
```

Going to rework this if you agree.

Reply:
Sounds fine to me -- the main thing I was interested in doing is to make sure we return the list of choices formatted like `[('A', 'A')]` from the API since that's generally what the client expects for these. Feel free to adjust how we get there, though.

Reply:
Will do